shopfloor_reception_purchase_partner_ref/actions/schema.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.addons.component.core import Component
+
+
+class ShopfloorSchemaAction(Component):
+    _inherit = "shopfloor.schema.action"
+
+    def picking(self):
+        res = super().picking()
+        res.update({"purchase_order": self._schema_dict_of(self.purchase_order())})
```

Comment:
This will be used by any picking parser.
I'd move move these changes to a base module for purchase (eg: `shopfloor_purchase_base`) that can be reused  and here add a conditional argument like `with_purchase_order` to be used only when needed.
WDYT?

Reply:
with_purchase_order would be great. 
But then we should also change the property _picking_parser and the schema for picking. 
![image](https://github.com/OCA/wms/assets/78783475/23c0ce13-f854-4380-8683-df8bda1150ff)
![image](https://github.com/OCA/wms/assets/78783475/47b7b1b7-7930-4f80-a9f5-b8768e623aa2)

and also we have to change the shopfloor_reception to call of those methods with "with_purchase_order"
![image](https://github.com/OCA/wms/assets/78783475/ff68e0a7-37dd-4394-928d-af6516a93ed0)

I think it would much easier, then to always add the purchase order to the picking.